chia/server/node_discovery.py

```diff
@@ -359,11 +358,13 @@ async def _connect_to_peers(self, random) -> None:
                     if local_peerinfo is not None and addr == local_peerinfo:
                         continue
                     got_peer = True
+                    self.log.debug(f"Addrman selected address: {addr}.")
 
                 disconnect_after_handshake = is_feeler
                 if self._num_needed_peers() == 0:
```

Comment:
we call `_num_needed_peers()` 3 times here. each time we traverse all our connection objects. Not a big deal, but seems a bit unnecessary.